packages/astro/src/core/messages.ts

```diff
@@ -116,6 +114,37 @@ export function devStart({
 	return messages.map((msg) => `  ${msg}`).join('\n');
 }
 
+export function telemetryNotice() {
+	const headline = yellow(`Astro now collects ${bold('anonymous')} usage data.`);
+	const why = `This ${bold('optional program')} will help shape our roadmap.`;
+	const more = `For more info, visit ${underline('https://astro.build/telemetry')}`;
+	const box = boxen([headline, why, '', more].join('\n'), {
+		margin: 0,
+		padding: 1,
+		borderStyle: 'round',
+		borderColor: 'yellow',
+	});
+	return box;
+}
+
+export function telemetryEnabled() {
+	return `\n  ${green('◉')} Telemetry is ${bgGreen(
```

Comment:
nit: slight preference for mentioning the word "anonymous" in both enabled and disabled messages. Something like:

`Anonymous telemetry enabled. Thank you for helping to improve Astro!`
`Anonymous telemetry disabled.`